From 91ad9d098a626b24655ece2c6f29e0f7a2849268 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Sat, 20 Nov 2021 04:02:39 +0200
Subject: [PATCH] [SW-60588] gaudi2: sec: add bad access for RR

In order for validation to monitor razwi occurrence during
a RR violation, bad access bit needs to be added in the
monitored bit list.

NOTE: This is a revert me patch and the logic will be removed
post Gaudi2 PO validation.

Change-Id: I8a86cd22152cd2366aafeceb456bf813b56a659f
---

diff --git a/arch/gaudi2/common/common.c b/arch/gaudi2/common/common.c
index 465d1ed..a52de69 100644
--- a/arch/gaudi2/common/common.c
+++ b/arch/gaudi2/common/common.c
@@ -1031,7 +1031,7 @@
 void check_razwi_intr(void)
 {
 	u32 reg = regmap_readl(mmPCIE_WRAP_PCIE_IC_SEI_INTR_IND);
-	u32 bitmask = reg & PCIE_AXI_ERR_INTR_BIT;
+	u32 bitmask = reg & PCIE_INTR_ERR_INTR_BIT;
 
 	if (bitmask) {
 		regmap_writel(RAZWI_INTR_STATUS_PATTERN, mmGIC_RAZWI_STATUS_REG);
diff --git a/arch/gaudi2/common/common.h b/arch/gaudi2/common/common.h
index 1be3149..d3e1b66 100644
--- a/arch/gaudi2/common/common.h
+++ b/arch/gaudi2/common/common.h
@@ -62,7 +62,7 @@
  * TODO: These are temporary macros
  * and to be removed after GAUDI2 PO
  */
-#define PCIE_AXI_ERR_INTR_BIT		(BIT(0) | BIT(1))
+#define PCIE_INTR_ERR_INTR_BIT		(BIT(0) | BIT(1) | BIT(2))
 #define RAZWI_INTR_STATUS_PATTERN	0xABBA
 #define NUM_OF_MME_WB_PORTS		2
 
